Cognite.Simulator.Tests/UtilsTests/TestUtilities.cs

```diff
@@ -101,24 +102,67 @@ public static Func<HttpRequestMessage, CancellationToken, Task<HttpResponseMessa
                     throw new ArgumentNullException(nameof(uri));
                 }
 
-                foreach (var mapping in endpointMappings)
+                foreach (var mockTemplate in endpointMockTemplates)
                 {
-                    if (mapping.Key(uri))
+                    if (mockTemplate.Matches(uri))
                     {
-                        var (responseFunc, callCount, maxCalls) = mapping.Value;
-                        if (maxCalls.HasValue && callCount >= maxCalls)
-                        {
-                            return GoneResponse;
-                        }
-                        endpointMappings[mapping.Key] = (responseFunc, callCount + 1, maxCalls);
-                        return responseFunc();
+                        return mockTemplate.GetResponse();
                     }
                 }
-
-                return new HttpResponseMessage(HttpStatusCode.OK) { Content = new StringContent("{\"items\":[]}") };
+                return CreateResponse(HttpStatusCode.NotImplemented, "Not implemented");
             };
         }
 
+        /// <summary>
+        /// A container for easier mocking of endpoints.
+        /// </summary>
+        public class SimpleRequestMocker
+        {
+            private readonly Func<string, bool> _uriMatcher;
+            private readonly Func<HttpResponseMessage> _responseFunc;
+            private int _callCount = 0;
+            private readonly int? _maxCalls = null;
+
+            /// <summary>
+            /// Creates a new SimpleRequestMocker.
+            /// </summary>
+            /// <param name="uriMatcher">Function to match the URI of the request.</param>
+            /// <param name="responseFunc">Function to generate the response.</param>
+            /// <param name="maxCalls">Maximum number of times this mocker can be called. If null, there is no limit.</param>
+            public SimpleRequestMocker(Func<string, bool> uriMatcher, Func<HttpResponseMessage> responseFunc, int? maxCalls = null)
+            {
+                _uriMatcher = uriMatcher;
+                _responseFunc = responseFunc;
+                _maxCalls = maxCalls;
+            }
+
+            private bool HasMoreCalls()
+            {
+                return !_maxCalls.HasValue || _callCount < _maxCalls;
+            }
+
+            public bool Matches(string uri)
+            {
+                return _uriMatcher(uri) && HasMoreCalls();
+            }
+
+            public HttpResponseMessage GetResponse()
+            {
+                _callCount++;
```

Comment:
should there be a guard here if we exceed the `maxCalls`?

Reply:
ok!